stub-idp/src/test/java/uk/gov/ida/stub/idp/repositories/jdbc/JDBIIdpSessionRepositoryTest.java

```diff
@@ -33,7 +34,26 @@ public void setUp() {
 		jdbi = Jdbi.create(DATABASE_URL);
 		repository = new JDBIIdpSessionRepository(jdbi);
 	}
-	
+
+	@Test
+	public void testSessionDeletion() {
+		// force clear all sessions from the test database
+		repository.deleteSessionsOlderThan(Duration.ZERO);
+		assertThat(repository.countSessionsInDatabase()).isZero().withFailMessage("found %d sessions in the database when it should be empty", repository.countSessionsInDatabase());
+		repository.createSession(createSession());
+		assertThat(repository.countSessionsInDatabase()).isEqualTo(1);
+		assertThat(repository.countSessionsOlderThan(Duration.standardHours(1))).isEqualTo(0);
+		assertThat(repository.countSessionsOlderThan(Duration.ZERO)).isEqualTo(1);
+		repository.deleteSessionsOlderThan(Duration.ZERO);
```

Comment:
I did think about adding a test for that but didn't because I was concerned about potential timing issues / adding sleep calls to a test.  

it would have to do something like:
* add a session
* sleep for x seconds
* add a session
* delete sessions older than x-1 seconds
* check only one session exists

I can add that but it might potentially be flaky unless x is ~3-5 seconds

Reply:
unless the code is changed to delete based on authnrequest issue instant - currently it's using last_modified as set by SessionRepositoryBase

Ah - and last_modified is not set when sessions are created...

Reply:
now last_modified should be set when sessions are created